src/lib.rs

```diff
@@ -436,21 +458,32 @@ pub trait Handler {
         Err(Error::Unimplemented)
     }
 
+    /// Return the identifier of the current thread.
     fn current_thread(&self) -> Result<Option<ThreadId>, Error> {
         Ok(None)
     }
 
+    /// Set the current thread for future operations.
     fn set_current_thread(&self, _id: ThreadId) -> Result<(), Error> {
         Err(Error::Unimplemented)
     }
 
+    /// Search memory.  The search begins at the given address, and
+    /// ends after length bytes have been searched.  If the provided
+    /// bytes are not seen, `None` should be returned; otherwise, the
+    /// address at which the bytes were found should be returned.
     fn search_memory(&self, _address: u64, _length: u64, _bytes: &[u8])
                      -> Result<Option<u64>, Error> {
         Err(Error::Unimplemented)
     }
 
+    /// Return the reason that the inferior has halted.
     fn halt_reason(&self) -> Result<StopReason, Error>;
 
+    /// Invoke a command.  The command is just a sequence of bytes
+    /// (typically ASCII characters), to be interpreted by the server
+    /// in any way it likes.  The result is output to send back to the
+    /// client.  This is used to implement gdb's `monitor` command.
```

Comment:
Would it be more useful for us to parse the command as ASCII and provide it as `&str`? Are there any legitimate cases where that's not valid?